core/vm/contracts.go

```diff
@@ -715,10 +753,99 @@ func (c *hashHeader) Run(input []byte, caller common.Address, evm *EVM, gas uint
 	var header types.Header
 	err = rlp.DecodeBytes(input, &header)
 	if err != nil {
-		return nil, gas, err
+		return nil, gas, ErrInputDecode
 	}
 
 	hashBytes := header.Hash().Bytes()
 
 	return hashBytes, gas, nil
 }
+
+type getParentSealBitmap struct{}
+
+func (c *getParentSealBitmap) RequiredGas(input []byte) uint64 {
+	return params.GetParentSealBitmapGas
+}
+
+// Return the signer bitmap from the parent seal of a past block in the chain.
+// Requested parent seal must have occurred within 4 epochs of the current block number.
+func (c *getParentSealBitmap) Run(input []byte, caller common.Address, evm *EVM, gas uint64) ([]byte, uint64, error) {
+	gas, err := debitRequiredGas(c, input, gas)
+	if err != nil {
+		return nil, gas, err
+	}
+
+	// input is comprised of a single argument:
+	//   blockNumber: 32 byte integer representing the block number to access
+	if len(input) < 32 {
+		return nil, gas, ErrInputLength
+	}
+
+	blockNumber := new(big.Int).SetBytes(input[0:32])
+
+	// Ensure the request is for information from a previously sealed block.
+	if blockNumber.Cmp(common.Big0) == 0 {
+		return nil, gas, ErrBlockNumberOutOfBounds
+	}
+	if blockNumber.Cmp(evm.Context.BlockNumber) >= 0 {
```

Comment:
Why is it invalid to ask for the parentseal bitmap of the current block?

Reply:
I was able to confirm that the parent seal is added in the block prepare step, which occurs before transactions are added to the block. https://github.com/celo-org/celo-blockchain/blob/fef30781c9215b3bba4422074dd9d38ed92aed92/miner/worker.go#L918

Reply:
So, I've changed it to allow getting the chain head parent seal.